src/auth/authenticator/tls/authenticator.go

```diff
@@ -0,0 +1,51 @@
+package tls
+
+import (
+	"crypto/x509"
+	"net/http"
+
+	"github.com/consensys/quorum-key-manager/pkg/errors"
+	"github.com/consensys/quorum-key-manager/src/auth/types"
+)
+
+const AuthMode = "Tls"
+
+type Authenticator struct {
+	Certificates []*x509.Certificate
+}
+
+func NewAuthenticator(cfg *Config) (*Authenticator, error) {
+	if len(cfg.Certificates) == 0 {
+		return nil, nil
+	}
+
+	auth := &Authenticator{Certificates: cfg.Certificates}
+
+	return auth, nil
+}
+
+// Authenticate checks certs and retrieve user Info
+// CN field -> Username
+// Organization -> Groups
+func (authenticator Authenticator) Authenticate(req *http.Request) (*types.UserInfo, error) {
+	// extract Certificate info from request if any
+	if len(req.TLS.PeerCertificates) == 0 {
```

Comment:
Should we add if `req.TLS != nil`? Looks like this may `panic` if TLS is off

Reply:
c.f. Kubernetes: https://github.com/kubernetes/kubernetes/blob/c224ca7674445ec98a2e8876daf53b9846cd3ae1/staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go#L134

Reply:
This was indeed an issue with http